django/contrib/admin/templatetags/admin_list.py

```diff
@@ -201,18 +203,19 @@ def link_in_col(is_first, field_name, cl):
         except ObjectDoesNotExist:
             result_repr = empty_value_display
         else:
-            empty_value_display = getattr(attr, 'empty_value_display', empty_value_display)
+            empty_value_display = mark_safe(getattr(attr, 'empty_value_display', empty_value_display))
```

Comment:
I don't see a reason why we should assume `empty_value_display` is safe. Patch to fix this and some other edits: http://dpaste.com/2TETT95
Does the new `allow_tags` test cover `django/contrib/admin/helpers.py` too? I don't think so as I think it's only part of the change form page (not change list).